task_manager/test_task.hh

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager.hh"
+
+class test_task : public task_manager::task::impl {
+public:
+    test_task(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_id parent_id = task_id::create_null_id()) 
+        : task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), parent_id)
+    {}
+
+    future<> abort() override {
+        return make_ready_future<>();
+    }
+
+    task_manager::task_ptr create_child(task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_id parent_id = task_id::create_null_id()) const override {
```

Comment:
Let's carry the sequence number from the parent.
It should dispatched only for the top level task, not for each child.
It would be nice to repeat it throughout the tree of tasks associated with the top one.
Kind of like a "job" sequence number...